addons/website_slides/static/src/js/slides_upload.js

```diff
@@ -53,32 +55,165 @@ var SlideDialog = Widget.extend({
-    _checkUniqueSlide: function (fileName) {
+    _bindSelect2Dropdown: function () {
+        var self = this;
+        this.$('#category_id').select2(this._select2Wrapper(_t('Category'), false,
+            function () {
+                return self._rpc({
+                    route: '/slides/category/search_read',
+                    params: {
+                        fields: ['name'],
+                        domain: [['channel_id', '=', self.channelID]],
+                    }
+                });
+            })
+        );
+        this.$('#tag_ids').select2(this._select2Wrapper(_t('Tags'), true, function () {
+            return self._rpc({
+                route: '/slides/tag/search_read',
+                params: {
+                    fields: ['name'],
+                    domain: [],
+                }
+            });
+        }));
+    },
+    _fetchUrlPreview: function (url) {
         return this._rpc({
-            model: 'slide.slide',
-            method: 'search_count',
-            args: [[['channel_id', '=', self.channelId], ['name', '=', fileName]]],
+            route: '/slides/prepare_preview/',
+            params: {
+                'url': url,
+                'channel_id': this.channelID
+            },
         });
     },
+    _formSetFieldValue: function (fieldName, value) {
+        this.$('form').find('#'+fieldName).val(value);
+    },
+    _formGetFieldValue: function (fieldName) {
+        return this.$('#'+fieldName).val();
+    },
+    _formValidate: function () {
+        var form = this.$("form");
+        form.addClass('was-validated');
+        return form[0].checkValidity() && this.is_valid_url;
+    },
     /**
+     * Extract values to submit from form, force the slide_type according to
+     * filled values.
+     *
      * @private
      */
-    _resetFile: function () {
+    _formValidateGetValues: function (force_published) {
+        var canvas = this.$('#data_canvas')[0];
+        var values = _.extend({
+            'channel_id': this.channelID,
+            'name': this._formGetFieldValue('name'),
+            'url': this._formGetFieldValue('url'),
+            'description': this._formGetFieldValue('description'),
+            'index_content': this._formGetFieldValue('index_content'),
+            'website_published': force_published,
+        }, this._getSelect2DropdownValues()); // add tags and category
+
+        // default slide_type (for webpage for instance)
+        if (_.contains(this.slide_type_data), this.get('state')) {
+            values['slide_type'] = this.get('state');
+        }
+
+        if (this.file.type === 'application/pdf') {
+            _.extend(values, {
+                'image': canvas.toDataURL().split(',')[1],
+                'slide_type': canvas.height > canvas.width ? 'document' : 'presentation',
+                'mime_type': this.file.type,
+                'datas': this.file.data
+            });
+        }
+        if (/^image\/.*/.test(this.file.type)) {
```

Comment:
I don't think a file can be 'application/pdf' AND 'image'.
I would make that a "else if" for readability.